sneaker/src/main/java/com/irozon/sneaker/Sneaker.java

```diff
@@ -189,24 +193,53 @@ public Sneaker setMessage(String message, int color) {
      * @return
      */
     public Sneaker setIcon(int icon) {
+        mIconDrawable = null;
         mIcon = icon;
         return this;
     }
 
     /**
+     * Sets the icon to sneaker
+     *
+     * @param icon Icon drawable for sneaker
+     * @return
+     */
+    public Sneaker setIcon(Drawable icon) {

```

Comment:
Love this idea, but you should check for a null icon. Or add @NonNull annotation to reduce exceptions